sql/catalyst/src/main/scala/org/apache/spark/sql/types/LongType.scala

```diff
@@ -43,7 +43,7 @@ class LongType private() extends IntegralType {
    */
   override def defaultSize: Int = 8
 
-  override def simpleString: String = "bigint"
+  override def simpleString: String = "long"
```

Comment:
You're renaming the whole type, when the issue is at best that some intermediate stage of the plan does cast to bigint first. At the least, this is not the fix, and I'm not sure it's really a problem, even if the error is from an implementation detail.